subsys/usb/usb_device.c

```diff
@@ -549,18 +549,29 @@ static bool usb_set_configuration(u8_t config_index, u8_t alt_setting)
 static bool usb_set_interface(u8_t iface, u8_t alt_setting)
 {
 	const u8_t *p = usb_dev.descriptors;
-	u8_t cur_iface = 0xFF;
+	const u8_t *if_desc = NULL;
 	u8_t cur_alt_setting = 0xFF;
-	struct usb_dc_ep_cfg_data ep_cfg;
+	u8_t cur_iface = 0xFF;
```

Comment:
Is this the network interface index value or some other value that is specific to USB? The reason I am asking, is that currently network interface index numbering starts from 1, and 0 value can be used to mean unknown or unset value.

Actually it is a bit confusing to read this commit as typically iface is a pointer to struct net_if and if_index, if_idx or similar is the network interface index value. Now if_desc is a pointer and iface is index, so reviewing this needs some more effort.

Reply:
This is not network interface, this is USB interface number.

Reply:
@jukkar It is specific to USB, means usb device interface (usb_device contains usb device code only)